### DISCUSSION TITLE
Is modifying snippets good practice?

### DISCUSSION THREAD

**Discussion by y-wertz-priomold** [chakra-ui/chakra-ui#8993] [Q&A] [answered]
Migrating to V3 introduced snippets (using `npx @chakra-ui/cli snippet add`) which are created in the `@/components/ui` folder.

I was wondering if these snippets are inteded to be modified by the developer without risking to be overwritten in the future by an update?

Or should I copy the content and save a modified version in another place? 
Problem with this is the need to update all imports already in use.. Also the doesn't prevent using the default snippet accidentally.

Thanks for help 😊

**Comment by segunadebayo** [comment, marked as answer]
Yes, please feel free to change them as you need to.

The CLI will skip existing files when you rerun it. We'll keep a changelog of updated snippets so you can either modify them, or override them.

**Reply by veloware** [reply]
@segunadebayo I think i've noticed that if you add an individual snippet, make some changes to it, then at some stage later run 

```
npx @chakra-ui/cli snippet add
```

(because you've just realised. that you need a lot of more snippets than first realised :D ) it will revert your changes made to the originally added individual snippet(s). Is this expected? Thanks!

**Reply by joekrill** [reply]
I've just noticed this as well. @segunadebayo is this a bug that should be reported? The CLI definitely did not skip existing files. And honestly I don't think that's necessarily _wrong_. In many cases I would _want_ snippets to be "upgraded". For example, the Slider component was fixed recently by updating the snippet. But in other cases I do _not_ want it to: I've modified the `provider.tsx` snipper to use my custom theme (am I doing this wrong? it's not possible to override it otherwise).
